runtime/near-vm-logic/src/logic.rs

```diff
@@ -828,9 +829,8 @@ impl<'a> VMLogic<'a> {
     /// * If passed gas amount somehow overflows internal gas counters returns `IntegerOverflow`;
     /// * If we exceed usage limit imposed on burnt gas returns `GasLimitExceeded`;
     /// * If we exceed the `prepaid_gas` then returns `GasExceeded`.
-    pub fn gas(&mut self, gas_amount: u32) -> Result<()> {
```

Comment:
Seems the name was misleading before, as otherwise we shall fix the call site as well. If what we're saving is mostly this multiplication - cannot we do multiplication when instrumenting the contract? Or it's a bad idea, as gas cost may change without contract recompilation?

Reply:
I think it's possible to do, but need to move it towards `u64`. But we also need to deduct and check the remaining gas. If we split burn gas and used gas, then it should not be such a large difference in performance comparing to this implementation.